enhancements/network/ingress-node/ingress-node-firewall-with-ebpf.md

```diff
@@ -0,0 +1,355 @@
+}
+
+// NodeIngressFirewallConfigConfigSpec consists of a node name which the private IP should be assigned to.
+// +k8s:openapi-gen=true
+type NodeIngressFirewallConfigSpec struct {
+	// A list of interfaces where the ingress firewall policy will be applied on.
+	// empty list indicates the firewall policy applied on all interfaces
+	// +kubebuilder:validation:Optional
+	// +optional
+	Interface []string `json:"interface" protobuf:"bytes,1,opt,name=interface"`
+
+	// A list of ingress firewall policy rules.
+	// empty list indicates no ingress firewall i.e allow all incoming traffic.
+	// +kubebuilder:validation:Optional
+	// +optional
+	Ingress []IngressNodeFirewallRules `json:"ingress" protobuf:"bytes,2,opt,name=ingress"`
+}
+
+// NodeIngressFirewallConfigConfigStatus specifies the node assignment together with its assignment condition.
+// +k8s:openapi-gen=true
+type NodeIngressFirewallConfigStatus struct {
+	// +kubebuilder:validation:Optional
+	// +optional
+}
+
+// Compatibility level 1: Stable within a major release for a minimum of 12 months or 3 minor releases (whichever is longer).
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:openapi-gen=true
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:path=nodeingressfirewallconfigs,scope=Cluster
+// +openshift:compatibility-gen:level=1
+type NodeIngressFirewallConfig struct {
+metav1.TypeMeta   `json:",inline"`
+metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+// spec is the definition of the desired private IP request.
+// +kubebuilder:validation:Required
+// +required
+Spec NodeIngressFirewallConfigSpec `json:"spec" protobuf:"bytes,2,opt,name=spec"`
+// status is the observed status of the desired private IP request. Read-only.
+// +kubebuilder:validation:Optional
+// +optional
+Status NodeIngressFirewallConfigStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
+}
+
+// Compatibility level 1: Stable within a major release for a minimum of 12 months or 3 minor releases (whichever is longer).
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +resource:path=nodeingressfirewallconfig
+// NodeIngressFirewallConfigList is the list of NodeIngressFirewallConfigList.
+// +openshift:compatibility-gen:level=1
+type NodeIngressFirewallConfigList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	// List of NodeIngressFirewallConfig.
+	Items []NodeIngressFirewallConfig `json:"items" protobuf:"bytes,2,rep,name=items"`
+}
+```
+
+### Test Plan
+
+- Unit tests coverage
+- IPv4/IPv6 E2E coverage to validate ingress node firewall functionality on OCP cluster with multiple external traffic streams to match allow and deny actions and verify statistics per rule.
+
+### Risks and Mitigations
+
+N/A
```

Comment:
This would be a good place to talk about the admin getting locked out by bad rules, and what we might do about it